src/sentry/grouping/ingest/config.py

```diff
@@ -27,7 +38,9 @@ def _project_should_update_grouping(project: Project) -> bool:
     should_update_org = (
         project.organization_id % 1000 < float(settings.SENTRY_GROUPING_AUTO_UPDATE_ENABLED) * 1000
     )
-    return bool(project.get_option("sentry:grouping_auto_update")) and should_update_org
+    deprecated_config = project.get_option("sentry:grouping_config") in CONFIGS_TO_DEPRECATE
```

Comment:
This code is reached when we save an error event, thus, we will also upgrade during ingestion:
https://github.com/getsentry/sentry/blob/233ccd6aa3c807f6a4085ded5c191f761f7485fd/src/sentry/event_manager.py#L1400